src/common/CudaDataUtils.hpp

```diff
@@ -42,6 +42,36 @@ __device__ __forceinline__ unsigned long long device_timer()
   return global_timer;
 }
 
+/*!
+ * \brief Method for launching a CUDA kernel with given configuration.
+ *
+ *        Note: method checks whether number of args and their types in 
+ *              kernel signature matches args passed to this method.
+ */
+template <typename... Args, typename...KernArgs>
+void RPlaunchCudaKernel(void (*kernel)(KernArgs...),
+                        const dim3& numBlocks, const dim3& dimBlocks,
+                        std::uint32_t sharedMemBytes, cudaStream_t stream,
+                        Args const&... args)
+{
+  static_assert(sizeof...(KernArgs) == sizeof...(Args),
+                "Number of kernel args doesn't match what's passed to method");
+
+  int ia[] = {[](){
+    static_assert(std::is_same<std::decay_t<KernArgs>, std::decay_t<Args>>::value, "Kernel arg types don't match what's passed to method");
+    return 0;
+  }()...};
+  RAJA_UNUSED_VAR(ia);
```

Comment:
We can use std::conjunction to check that all of the conditions are true in a single static_assert. It is c++17 but it can be implemented in c++14, see possible implementation here https://en.cppreference.com/w/cpp/types/conjunction.
```suggestion
    static_assert(conjunction<std::is_same<std::decay_t<KernArgs>, std::decay_t<Args>>...>::value, "Kernel arg types don't match what's passed to method");
```